addon/convert-schema.js

```diff
@@ -0,0 +1,200 @@
+    if (aggregateType !== undefined) {
+      retModel[aggregateType] = _.map(model[aggregateType], (subSchema) => {
+        return convertConditionalProperties(subSchema, value, getPreviousValue)
+      })
+    } else if (model.not) {
+      retModel.not = convertConditionalProperties(model.not, value, getPreviousValue)
+    }
+  }
+
+  let depsMet = {}
+  let props = {}
+
+  const getValue = pathFinder(value, getPreviousValue)
+
+  _.each(retModel.properties, function (subSchema, propName) {
+    retModel.properties[propName] = convertConditionalProperties(subSchema, _.get(value, propName), pathFinder(value, getValue))
+  })
+  let conditionalProperties = _.transform(model.properties, function (result, schema, key) {
+    if (schema.conditions) {
+      result[key] = schema
+    }
+  })
+  _.each(conditionalProperties, function (depSchema, key) {
+    depsMet[key] = _.some(depSchema.conditions, function (enableConditions) {
+      const hasDependencyMet = _.some(enableConditions.if, function (conditionList) {
+        return _.every(conditionList, function (conditionValue, dependencyKey) {
+          const dependencyValue = getValue(dependencyKey)
+          return meetsCondition(dependencyValue, conditionValue)
+        })
+      })
+      if (hasDependencyMet && enableConditions.then !== undefined) {
+        props[key] = _.cloneDeep(enableConditions.then)
+      }
+      return hasDependencyMet
+    })
+  })
+  _.each(depsMet, function (dependencyMet, depName) {
+    const baseSchema = retModel.properties[depName]
+    if (dependencyMet && !baseSchema.set || !dependencyMet && baseSchema.set) {
+      retModel.properties[depName] = _.omit(_.defaults(props[depName] || {}, baseSchema), ['conditions', 'set'])
+    } else {
+      delete retModel.properties[depName]
+    }
+  })
+
+  return retModel
+}
+
+export default convertConditionalProperties
+
+export function convertView (model, view) {
+  function getSchema (pathStack, model) {
+    if (pathStack.length <= 0 || model === undefined) {
+      return model
+    }
+
+    if (model.properties) {
+      const current = pathStack.pop()
+      return getSchema(pathStack, model.properties[current])
+    }
+
+    if (model.items) {
+      return getSchema(pathStack, model.items)
+    }
+  }
+
+  const parentContainers = {}
+  _.each(view.containers, function (container, index) {
+    _.each(container.rows, function (row, rowIndex) {
+      _.each(row, function (rowCell, rowCellIndex) {
+        const subContainer = _.get(rowCell, 'container') || _.get(rowCell, 'item.container')
+        if (subContainer) {
+          parentContainers[subContainer] = {container: container.id, rowIndex, rowCellIndex}
+        }
+      })
+    })
+  })
+
+  function getNextContainer (containerData) {
+    if (containerData) {
+      return _.find(view.containers, function (container) {
+        return container.id === containerData.container
+      })
+    }
+  }
```

Comment:
Not a huge fan of early returns. I also tend not to use fat arrows when not necessary because it helps me avoid relying on `this`.